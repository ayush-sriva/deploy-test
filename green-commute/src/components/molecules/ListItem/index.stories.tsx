import { ComponentMeta, ComponentStory } from '@storybook/react';
import Olaurl from '../../../../public/assets/logos/ola.png';
import Rapido from '../../../../public/assets/logos/rapido.png';
import Uber from '../../../../public/assets/logos/uber.png';
import ListItem from '.';

export default {
  title: 'components/Molecules/ListItem',
  component: ListItem,
} as ComponentMeta<typeof ListItem>;

const Template: ComponentStory<typeof ListItem> = (args) => <ListItem {...args} />;

export const OlaListItem = Template.bind({});

OlaListItem.args = {
  logoUrl: `${Olaurl}`,
  company: 'Ola',
  approx: 'Approximately  ',
  fare: '45',
};

export const RapidoListItem = Template.bind({});

RapidoListItem.args = {
  logoUrl: `${Rapido}`,
  company: 'Ola',
  approx: 'Approximately  ',
  fare: '45',
};

export const UberListItem = Template.bind({});

UberListItem.args = {
  logoUrl: `${Uber}`,
  company: 'Ola',
  approx: 'Approximately  ',
  fare: '45',
};
