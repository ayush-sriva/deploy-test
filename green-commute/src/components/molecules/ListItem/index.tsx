/**
 * @author Abhishek Bhemisetty <abhishek.bhemisetty@zemosolabs.com>
 */

import Grid from '@mui/material/Grid';
import Logo from '../../atoms/Logo';
import Typography from '../../atoms/Typography';
import theme from '../../../theme';
import Paper from '@mui/material/Paper';
import styled from 'styled-components';
import makeStyles from '@mui/styles/makeStyles';

interface ListProps {
  logoUrl: string;
  company: string;
  approx: string;
  fare: string;
}

const useStyles = makeStyles({
  paper: {
    backgroundColor: `${theme.palette.primary.contrastText}`,
    maxWidth: '350px',
    flexGrow: 1,
    height: '60px',
    marginBottom: '8px',
  },
  grid: {
    textAlign: 'center',
  },
  colorText: {
    color: `${theme.palette.primary.main}`,
  },
});

const ListItem: React.FC<ListProps> = ({ logoUrl, company, approx, fare }) => {
  const classes = useStyles();
  return (
    <Paper elevation={1} className={classes.paper}>
      <Grid container spacing={1} justifyContent='space-between' alignItems='center' data-testid='main-grid'>
        <Grid className={classes.grid} item xs={1.5} data-testid='logo-grid'>
          <Logo url={logoUrl} />
        </Grid>
        <Grid item xs={7} sm>
          <Grid container direction='column' justifyContent='space-between' spacing={3}>
            <Grid container item xs={6}>
              <Typography variant='captionBold'>{company}</Typography>
            </Grid>
            <Grid container item xs={6}>
              <Typography variant='caption'>
                {approx} â‚¹{fare}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={3} data-testid='book-now'>
          <Typography variant='captionBold' className={classes.colorText}>
            Book Now
          </Typography>
        </Grid>
      </Grid>
    </Paper>
  );
};

export default ListItem;
