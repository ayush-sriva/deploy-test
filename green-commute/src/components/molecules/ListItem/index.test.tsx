import { render, screen } from '@testing-library/react';
import ReactTestRenderer from 'react-test-renderer';
import ListItem from '.';

describe('List Item molecule', () => {
  test('Ola list item molecule', () => {
    render(
      <ListItem
        logoUrl='../../../../public/assets/logos/ola.png'
        company='Ola'
        approx={''}
        fare={''}
      ></ListItem>
    );
    expect(screen.getByText('Ola')).toBeInTheDocument();
    expect(screen.getByTestId('book-now')).toBeInTheDocument();
  });

  test('Rapido list item molecule with approx and fare', () => {
    render(
      <ListItem
        logoUrl='../../../../public/assets/logos/rapido.png'
        company='Rapido'
        approx='Approximately'
        fare='70'
      ></ListItem>
    );
    expect(screen.getByText(/Approximately/)).toBeInTheDocument();
    expect(screen.getByText(/70/)).toBeInTheDocument();
  });

  test('Uber list item molecule logo rendering test', () => {
    render(
      <ListItem
        logoUrl='../../../../public/assets/logos/uber.png'
        company='Uber'
        fare='30'
        approx={''}
      ></ListItem>
    );
    expect(screen.getByTestId('logo-grid')).toBeInTheDocument();
  });
});

describe('List Item Snapshot', () => {
  it('renders list item correctly', () => {
    const tree = ReactTestRenderer.create(
      <ListItem
        logoUrl='../../../../public/assets/logos/rapido.png'
        company='Rapido'
        approx='Approximately'
        fare='70'
      ></ListItem>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
