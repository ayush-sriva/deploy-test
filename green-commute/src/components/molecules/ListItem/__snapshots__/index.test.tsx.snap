// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Snapshot renders list item correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-10 css-1ps6pg7-MuiPaper-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-15hauir-MuiGrid-root"
    data-testid="main-grid"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5 makeStyles-grid-11 css-1duidt6-MuiGrid-root"
      data-testid="logo-grid"
    >
      <img
        alt="Logo"
        data-testid="logo"
        src="../../../../public/assets/logos/rapido.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-true css-unluoz-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-1oolkk7-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
          >
            Rapido
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            Approximately
             â‚¹
            70
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      data-testid="book-now"
    >
      <span
        className="MuiTypography-root MuiTypography-captionBold makeStyles-colorText-12 css-37jdci-MuiTypography-root"
      >
        Book Now
      </span>
    </div>
  </div>
</div>
`;
